pkg/daemon/update.go

```diff
@@ -579,6 +579,21 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		return err
 	}
 
+	defer func() {
+		if retErr != nil {
+			if err := dn.updateSSHKeys(oldIgnConfig.Passwd.Users); err != nil {
+				errs := kubeErrs.NewAggregate([]error{err, retErr})
+				retErr = fmt.Errorf("error rolling back password hash updates: %w", errs)
+				return
+			}
+		}
+	}()
+
+	// Set password hash
+	if err := dn.SetPasswordHash(newIgnConfig.Passwd.Users); err != nil {
+		return err
+	}
+
 	defer func() {
 		if retErr != nil {
 			if err := dn.updateSSHKeys(oldIgnConfig.Passwd.Users); err != nil {
```

Comment:
Should this defer call SetPasswordHash() ?
Looks to me that ordering is a bit messed up for update SSH Key and set  password